clang/test/CIR/CodeGen/if-consteval.cpp

```diff
@@ -0,0 +1,50 @@
+// RUN: %clang_cc1 -std=c++23 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s --check-prefix=CIR
+// RUN: %clang_cc1 -std=c++23 -triple x86_64-unknown-linux-gnu -fclangir -emit-llvm %s -o %t-cir.ll
+// RUN: FileCheck --input-file=%t-cir.ll %s --check-prefix=LLVM
+// RUN: %clang_cc1 -std=c++23 -triple x86_64-unknown-linux-gnu -emit-llvm %s -o %t.ll
+// RUN: FileCheck --input-file=%t.ll %s --check-prefix=OGCG
+
+void f() {
+  int result = 0;
+
+  if consteval {
+    result = 10; 
+    // CIR-NOT: cir.const #cir.int<10> : !s32i
+    // LLVM-NOT: store i32 10, ptr %1, align 4
```

Comment:
```suggestion
    // LLVM-NOT: store i32 10
```
Similar changes should be made for the other "NOT" checks. The "NOT" checks should be as limited as possible to be sure the statement isn't generated. For instance, in this case if the compiler generated `store i32 10, ptr %2, align 4` the test would pass because the difference in variable name would cause the line not to be matched.

Reply:
🤯🤯 I hadn’t thought of that